samples/cpp/image.cpp

```diff
@@ -27,8 +27,13 @@ static void help()
 
 int main( int argc, char** argv )
 {
-    help();
-    const char* imagename = argc > 1 ? argv[1] : "../data/lena.jpg";
+    cv::CommandLineParser parser(argc, argv, "{help h ||}{@image|../data/lena.jpg|}");
+    if (parser.has("help"))
+    {
+        help();
+        return 0;
+    }
+    const char* imagename = parser.get<string>("@image").c_str();
```

Comment:
This line is wrong, because temporary string object will be destroyed and `imagename` will point to invalid memory address.
Type should be changed to `string` and some calls to `.c_str()` should be added in places were C-style array is needed.